test/reference-test/first-record.ts

```diff
@@ -12,72 +12,72 @@ export function getFirstRecord(filename: string): [true, any] | [false, null] {
       return [
         true,
         {
-          bigint_col: '0',
+          bigint_col: 0n,
           bool_col: true,
-          date_string_col: '01/01/09',
-          double_col: '0',
-          float_col: '0',
-          id: '0',
-          int_col: '0',
-          smallint_col: '0',
-          string_col: '0',
-          timestamp_col: '0',
-          tinyint_col: '0',
+          date_string_col: Buffer.from('01/01/09'),
+          double_col: 0,
+          float_col: 0,
+          id: 0,
+          int_col: 0,
+          smallint_col: 0,
+          string_col: Buffer.from('0'),
+          timestamp_col: 0,
+          tinyint_col: 0,
         },
       ];
     case 'alltypes_plain.parquet':
       return [
         true,
         {
-          bigint_col: '0',
+          bigint_col: 0n,
           bool_col: true,
-          date_string_col: '03/01/09',
-          double_col: '0',
-          float_col: '0',
-          id: '4',
-          int_col: '0',
-          smallint_col: '0',
-          string_col: '0',
-          timestamp_col: '0',
-          tinyint_col: '0',
+          date_string_col: Buffer.from('03/01/09'),
+          double_col: 0,
+          float_col: 0,
+          id: 4,
+          int_col: 0,
+          smallint_col: 0,
+          string_col: Buffer.from('0'),
+          timestamp_col: 0,
+          tinyint_col: 0,
         },
       ];
     case 'alltypes_plain.snappy.parquet':
       return [
         true,
         {
-          bigint_col: '0',
+          bigint_col: 0n,
           bool_col: true,
-          date_string_col: '04/01/09',
-          double_col: '0',
-          float_col: '0',
-          id: '6',
-          int_col: '0',
-          smallint_col: '0',
-          string_col: '0',
-          timestamp_col: '0',
-          tinyint_col: '0',
+          date_string_col: Buffer.from('04/01/09'),
```

Comment:
Using the rust based parquet CLI, looks like the schema for this is:
```
  OPTIONAL BYTE_ARRAY date_string_col;
```
whereas the `alltypes_tiny_pages.parquet` is 
```
  OPTIONAL BYTE_ARRAY date_string_col (STRING);
```
So the schemas for these similar test files aren't necessarily the same, and may explain some subtle differences.